easybuild/easyblocks/generic/tarball.py

```diff
@@ -57,8 +75,25 @@ def build_step(self):
 
     def install_step(self, src=None):
         """Install by copying from specified source directory (or 'start_dir' if not specified)."""
-        remove_dir(self.installdir)
-        copy_dir(src or self.cfg['start_dir'], self.installdir, symlinks=self.cfg['keepsymlinks'])
+
+        # Run preinstallopts before copy of source directory
+        if self.cfg['preinstallopts']:
+            preinstall_opts = self.cfg['preinstallopts'].split('&&')
```

Comment:
While this usage of `preinstallopts` is quite common, what happens here is reliant on the user knowing that.  As well as supporting this usage of `preinstallopts`, maybe we can primarily support an additional explicit parameter `preinstall_cmd` which we can explicitly document and attach this parsing of `preinstallopts` to it, that way you can support both cases. Something like
```
preinstall_cmd = None
if self.cfg['preinstallopts']:
    ...
    preinstall_cmd = '&&'.join([opt for opt in preinstall_opts if opt and not opt.isspace()])
if self.cfg['preinstall_cmd']:
    preinstall_cmd = ' '.join([preinstall_cmd, self.cfg['preinstall_cmd']])
if preinstall_cmd:
    self.log.info("Preparing installation of %s using command '%s'..." % (self.name, preinstall_cmd))
    run_cmd(preinstall_cmd, log_all=True, simple=True)
```

Reply:
I have to say that I always try first to use the different `opts` in the common steps of the build, but adding an explicit parameter is a good idea as well. I made it work in combination with `preinstallopts` as you suggested so that both options can be used at the same time.